sdks/python/apache_beam/pipeline.py

```diff
@@ -525,6 +525,14 @@ def run(self, test_runner_api='AUTO'):
     self.contains_external_transforms = (
         ExternalTransformFinder.contains_external_transforms(self))
 
+    self.contains_run_inference_transform = (
+        RunInferenceSideInputFinder.contains_run_inference_transform(self))
+
+    if (self.contains_run_inference_transform and
+        not self._options.view_as(StandardOptions).streaming):
+      raise RuntimeError(
```

Comment:
Could we please add a test to validate this gets correctly thrown?

Reply:
I am good with error too.

Reply:
I think I like the error that you have